lib/Runtime/Library/BuiltIn/BuiltIn.js

```diff
@@ -0,0 +1,60 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+"use strict";
+
+(function (intrinsic) {
+    var platform = intrinsic.BuiltIn;
+
+    let FunctionsEnum = {
+        ArrayIndexOf: { className: "Array", methodName: "indexOf", argumentsCount: 1 },
+    };
+
+    platform.registerFunction(FunctionsEnum.ArrayIndexOf, function (searchElement, fromIndex) {
```

Comment:
>FunctionsEnum.ArrayIndexOf [](start = 30, length = 26)

Maybe add a link to the spec section for ergonomics in referencing the implementation?

* [https://tc39.github.io/ecma262/#sec-array.prototype.indexof](https://tc39.github.io/ecma262/#sec-array.prototype.indexof)

In your first commit, the message said array+string.prototype.indexOf -- but the implementations are different, right? (It appears you're doing just Array here.)

[https://tc39.github.io/ecma262/#sec-string.prototype.indexof](https://tc39.github.io/ecma262/#sec-string.prototype.indexof)